packages/docusaurus-theme-common/src/utils/admonitionUtils.tsx

```diff
@@ -14,15 +14,21 @@ function extractMDXAdmonitionTitle(children: ReactNode): {
   rest: ReactNode;
 } {
   const items = React.Children.toArray(children);
-  const mdxAdmonitionTitle = items.find(
-    (item) =>
-      React.isValidElement(item) &&
-      (item.props as {mdxType: string} | null)?.mdxType ===
-        'mdxAdmonitionTitle',
-  );
+  const mdxAdmonitionTitle = items.find((item) => {
+    if (!React.isValidElement(item)) {
+      return false;
+    }
+    const props = item.props as {
+      mdxType: string;
+      'data-admonition-title'?: boolean;
+    } | null;
+    return (
+      props?.mdxType === 'template' && props['data-admonition-title'] === true
+    );
+  }) as JSX.Element | undefined;
   const rest = <>{items.filter((item) => item !== mdxAdmonitionTitle)}</>;
   return {
-    mdxAdmonitionTitle,
+    mdxAdmonitionTitle: mdxAdmonitionTitle?.props.children,
```

Comment:
Hey

Actually the template trick seems un-necessary: you just found the issue here

See prod markup:

https://docusaurus.io/tests/pages/markdownPageTests#admonitions

![CleanShot 2022-10-26 at 17 58 28@2x](https://user-images.githubusercontent.com/749374/198076109-54f8666c-868b-482e-9d98-05ea90b842a4.png)


The utils code does not unwrap the element and still let it pass through up to the React reconcilier (which emits the warning)

Just adding this line fixes the issue

Reply:
Ah, great find!